src/ol/layer/layer.js

```diff
@@ -180,8 +180,9 @@ ol.layer.Layer.prototype.setMap = function(map) {
     this.mapPrecomposeKey_ = ol.events.listen(
         map, ol.render.EventType.PRECOMPOSE, function(evt) {
           var layerState = this.getLayerState();
+          var zIndex = this.getZIndex();
           layerState.managed = false;
-          layerState.zIndex = Infinity;
+          layerState.zIndex = (goog.isDef(zIndex) && zIndex !== 0) ? zIndex : Infinity;
```

Comment:
Curious why `0` wouldn't be a valid z-index here.

As above, please do not use `goog.isDef()`.

Reply:
I based my code on the pull request #4298, and thought it had to be that way. Should I just allow 0 as an index then?